src/transforms/groupby.js

```diff
@@ -163,12 +177,14 @@ function transformOne(trace, state) {
 
         var newTrace = newData[i] = Lib.extendDeepNoArrays({}, trace);
 
+        cloneTransforms(newTrace);
+
         arrayAttrs.forEach(initializeArray.bind(null, newTrace));
 
-        for(var j = 0; j < len; j++) {
+        for(j = 0; j < len; j++) {
             if(groups[j] !== groupName) continue;
 
-            arrayAttrs.forEach(pasteArray.bind(0, newTrace, trace, j));
+            arrayAttrs.forEach(pasteArray.bind(null, newTrace, trace, j));
```

Comment:
@alexcjohnson @etpinard I've refactored… actually a fair amount of groupby. It was mostly trivial reordering and inlining in order to cut down on function calls and `slice` and `concat` operations, just to streamline things a bit function-call and memory-wise.